### ISSUE TITLE
Explaining the modification of url provided to start-server-and-test

### ISSUE THREAD

**Issue by LuckyWindsck** [LuckyWindsck/japan#1]
Commit: `ci: fix npm script test:e2e` https://github.com/LuckyWindsck/japan/commit/e74cb278df71e0e35669fc04c7cbf32e7df5daa5

There was a problem that `test:e2e` would be hung forever.

In order to solve the problem that the npm script `test:e2e` couldn't be executed successfully, I found a workaround by providing another url to `start-server-and-test`.

So the situation is that, after running the command `vite dev --port 4173`, when I access `http://localhost:4173`, I will be redirected to `http://localhost:4173/japan/`. I guess that the `base` config in `vite.config.ts` does the job.

However, after running the command `yarn build && yarn preview`, when I access `http://localhost:4173`, this time it won't trigger any redirection.

I am not sure whether the current configuration and this behavior are correct or not, but this brings the problem that the npm script `test:e2e` would be hung forever.

I tried to debug `start-server-and-test` by running the following command:

```shell
$ DEBUG=start-server-and-test yarn start-server-and-test preview http://localhost:4173 'cypress run --e2e'
```

It shows that `wait-on`, the library that `start-server-and-test` uses, keeps making HTTP head request to url `http://localhost:4173` and recieving status code `404`.

Extraction of command output:
```
making HTTP(S) head request to  url:http://localhost:4173 ...
  HTTP(S) error for http://localhost:4173 AxiosError: Request failed with status code 404
```

<img width="1593" alt="image" src="https://github.com/LuckyWindsck/japan/assets/24631178/4b4fac82-1fcd-4168-bce6-176cf97516b4">

However, I found that when I modify the url for `wait-on` to either `tcp:4173` or `http://localhost:4173/japan`, `cypress run --e2e` would be executed normally.

This is because that when the url is `tcp:4173`, `wait-on` will attempt to make TCP connection instead of HTTP request, and obviously the connection will be success.

<img width="1599" alt="image" src="https://github.com/LuckyWindsck/japan/assets/24631178/d2685725-d4cb-4a74-95d9-7c5e7871df62">

On the other hand, when the url is `http://localhost:4173/japan`, the HTTP request will recieve status code `200`, and therefore cypress can be executed.

<img width="1479" alt="image" src="https://github.com/LuckyWindsck/japan/assets/24631178/1e2bfa57-105e-4814-84cc-56d07fa8f32b">

So basically I just need to change the url provided to `start-server-and-test`, or that said `wait-on`, I can solve this problem.

**Comment by LuckyWindsck** [comment]
Btw, the reason why I create an issue to explain the commit is that, the rule [body-max-length](https://commitlint.js.org/#/reference-rules?id=body-max-length) of `commitlint` stops me from creating a commit with super long commit message.
